libs/types/src/precinct_selection.ts

```diff
@@ -1,22 +1,65 @@
 import { z } from 'zod';
-import { Precinct } from './election';
-import { IdSchema } from './generic';
+import { Precinct, PrecinctId, PrecinctIdSchema } from './election';
 
-export enum PrecinctSelectionKind {
-  SinglePrecinct = 'SinglePrecinct',
-  AllPrecincts = 'AllPrecincts',
+export type PrecinctSelectionKind = 'SinglePrecinct' | 'AllPrecincts';
+export const PrecinctSelectionKindSchema: z.ZodSchema<PrecinctSelectionKind> =
+  z.union([z.literal('SinglePrecinct'), z.literal('AllPrecincts')]);
+
+export interface SinglePrecinctSelection {
+  kind: 'SinglePrecinct';
+  precinctId: PrecinctId;
 }
+export const SinglePrecinctSelectionSchema: z.ZodSchema<SinglePrecinctSelection> =
+  z.object({
+    kind: z.literal('SinglePrecinct'),
+    precinctId: PrecinctIdSchema,
+  });
 
-export const PrecinctSelectionKindSchema = z.nativeEnum(PrecinctSelectionKind);
+export interface AllPrecinctsSelection {
+  kind: 'AllPrecincts';
+}
+export const AllPrecinctsSelectionSchema: z.ZodSchema<AllPrecinctsSelection> =
+  z.object({
+    kind: z.literal('AllPrecincts'),
+  });
 
-export type PrecinctSelection =
-  | { kind: PrecinctSelectionKind.AllPrecincts }
-  | { kind: PrecinctSelectionKind.SinglePrecinct; precinctId: Precinct['id'] };
+export type PrecinctSelection = SinglePrecinctSelection | AllPrecinctsSelection;
 
 export const PrecinctSelectionSchema: z.ZodSchema<PrecinctSelection> = z.union([
-  z.object({ kind: z.literal(PrecinctSelectionKind.AllPrecincts) }),
-  z.object({
-    kind: z.literal(PrecinctSelectionKind.SinglePrecinct),
-    precinctId: IdSchema,
-  }),
+  SinglePrecinctSelectionSchema,
+  AllPrecinctsSelectionSchema,
 ]);
+
+export function getSinglePrecinctSelection(
+  precinctId: PrecinctId
+): SinglePrecinctSelection {
+  return {
+    kind: 'SinglePrecinct',
+    precinctId,
+  };
+}
+
+export const ALL_PRECINCTS_SELECTION: AllPrecinctsSelection = {
+  kind: 'AllPrecincts',
+};
+
+export const ALL_PRECINCTS_NAME = 'All Precincts';
+
+export function getPrecinctSelectionName(
```

Comment:
This might actually better fit in `libs/utils` - if we put it there, we can also make us of the `find` util instead of manually recreating that logic. The same could be said of `getSinglePrecinctSelection`

Reply:
Will do!